code/modules/mob/living/inventory.dm

```diff
@@ -180,3 +162,191 @@
 	if((src.l_hand && !( src.l_hand.abstract )) || (src.r_hand && !( src.r_hand.abstract )))
 		return 1
 	return 0
+
+// This handles the drag-open inventory panel.
+/mob/living/MouseDrop(atom/over_object)
+	var/mob/living/L = over_object
+	if(istype(L) && L != src && L == usr && Adjacent(L))
+		show_inventory_panel(L)
+	. = ..()
+
+/mob/living/proc/show_inventory_panel(mob/user, datum/tgui_state/state)
+	if(!inventory_panel_type)
+		return FALSE
+
+	if(!inventory_panel)
+		inventory_panel = new inventory_panel_type(src)
+	inventory_panel.tgui_interact(user, null, state)
+
+	return TRUE
+
+// TGUITODO: Don't forget to Destroy() these properly!
+/datum/inventory_panel
+	var/mob/living/host
+	var/tgui_id = "InventoryPanel"
+
+/datum/inventory_panel/New(mob/living/new_host)
+	if(!istype(new_host))
+		qdel(src)
+		return
+	host = new_host
+	. = ..()
+
+/datum/inventory_panel/Destroy()
+	host = null
+	. = ..()
+
+/datum/inventory_panel/tgui_host(mob/user)
+	return host.tgui_host()
+
+/datum/inventory_panel/tgui_state(mob/user)
+	return GLOB.tgui_physical_state
+
+/datum/inventory_panel/tgui_status(mob/user, datum/tgui_state/state)
+	if(!host)
+		return STATUS_CLOSE
+	if(isAI(user))
+		return STATUS_CLOSE
+	return ..()
+
+/datum/inventory_panel/tgui_interact(mob/user, datum/tgui/ui, datum/tgui_state/custom_state)
+	if(!host)
+		qdel(src)
+		return
+	// This looks kinda complicated, but it's just making sure that the correct state is definitely set
+	// before calling open(), so that there isn't any accidental UI closes
+	var/open = FALSE
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, tgui_id, host.name)
+		open = TRUE
+	if(custom_state)
+		ui.set_state(custom_state)
+	if(open)
+		ui.open()
+	return ui
+
+/datum/inventory_panel/tgui_data(mob/user, datum/tgui/ui, datum/tgui_state/state)
+	var/list/data = ..()
+
+	data["slots"] = list()
+	data["slots"].Add(list(list(
+		"name" = "Head (Mask)",
+		"item" = host.wear_mask,
+		"act" = "mask",
+	)))
+	data["slots"].Add(list(list(
+		"name" = "Left Hand",
+		"item" = host.l_hand,
+		"act" = "l_hand",
+	)))
+	data["slots"].Add(list(list(
+		"name" = "Right Hand",
+		"item" = host.r_hand,
+		"act" = "r_hand",
+	)))
+	data["slots"].Add(list(list(
+		"name" = "Back",
+		"item" = host.back,
+		"act" = "back",
+	)))
+	data["slots"].Add(list(list(
+		"name" = "Pockets",
+		"item" = "Empty Pockets",
+		"act" = "pockets",
+	)))
+
+	data["internals"] = host.internals
+	data["internalsValid"] = istype(host.wear_mask, /obj/item/clothing/mask) && istype(host.back, /obj/item/weapon/tank)
+
+	return data
+
+/datum/inventory_panel/tgui_act(action, list/params, datum/tgui/ui, datum/tgui_state/state)
+	if(..())
+		return TRUE
+
+	// If anyone wants the inventory panel to actually work, 
```

Comment:
I'm not really sure why non-human living animals even have an inventory panel. They don't have any corresponding Topic() to handle stripping or putting stuff on them. I guess it's just to view their slots?